# HG changeset patch
# User John Donoghue <john.donoghue@ieee.org>
# Date 1389662475 18000
#      Mon Jan 13 20:21:15 2014 -0500
# Branch gui-release
# Node ID 8449cc186059eed591f05a7f911f46ea775073b3
# Parent  f227bb47314fb6fcd5428bd6ac6df3bea407450b
Set terminal palette background color (Bug #41206)

* libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
  (QConsolePrivate::setBackgroundColor): Set color palette usingbackground color.

diff --git a/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp b/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
--- a/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
+++ b/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
@@ -570,16 +570,19 @@ QColor QConsolePrivate::selectionColor (
 QColor QConsolePrivate::cursorColor (void) const
 {
   return m_cursorColor.isValid () ? m_cursorColor : foregroundColor ();
 }
 
 void QConsolePrivate::setBackgroundColor (const QColor& color)
 {
   m_colors[15] = color;
+
+  QPalette palette (color);
+  m_consoleView->setPalette (palette);
 }
 
 void QConsolePrivate::setForegroundColor (const QColor& color)
 {
   m_colors[0] = color;
 }
 
 void QConsolePrivate::setSelectionColor (const QColor& color)
